### DISCUSSION TITLE
Disable inheritance on @embedded?

### DISCUSSION THREAD

**Discussion by eivihnd** [isar/isar#747] [Help] [answered]
TLDR: 
Is it possible to use "inheritance: false" on `@embedded`?

I am trying to follow clean architecture principles using the repository pattern, and implement Isar as my local cache in the data layer. Each dataModel in the data layer extends an Entity in the domain layer.

I have a `VideoModel` that extends the `Video` entity (which extends `Equatable`).
The `VideoModel` contains (amongst other things) a list of embedded `AssetModel`.
The `AssetsModel` extends the `Assets` entity (which extends `Equatable`).
An `AssetModel` contains (amongst other things) a `ImagesModel` member.
The `ImagesModel` extends the `Images` entity (which extends `Equatable`).

Now, If I mark the `AssetsModel` and `ImagesModel` with `@embedded`, I need to annotate the entity classes as well. To me, that clutters the domain layer with code specific to the data layer. I think I would be able to solve my problem if I could annotate `@embedded` classes with something like `@embedded(inheritance: false)`.

Is this a good idea, or are there something fundamental that I have't understood?

**Comment by simc** [comment, marked as answer]
You can do that using `@Embedded(inheritance: false)`

**Reply by eivihnd** [reply]
Aha! I was only looking at the isar.dev documentaion, and missed the pub.dev API documentation. My bad, sorry.
